internal/scaler/awsec2asg/aws_ec2_asg.go

```diff
@@ -319,7 +319,7 @@ func (s *Scaler) Register(ctx context.Context) error {
 
 			var updateAsg *autoscaling.UpdateAutoScalingGroupInput
 
-			desiredCapacity := inputData.DesiredCount
+			desiredCapacity := inputData.DesiredCount + *asg.DesiredCapacity
```

Comment:
We should try to avoid hard-coding our desired logic here and instead try setting `inputData.DesiredCount` to have the desired count.

so, the inputData to the scaler is being provided by the go template present in the connector config (in waymond.toml config file) that uses the ec2_aws_asg scaler. In that go template, we will need to make use of `{{ .scheduled_jobs_count + .running_jobs_count }}` instead of `{{ .scheduled_jobs_count }}` to set the desired count.

Reply:
we probably need to add support in buildkite trigger to output the running jobs count similar to how we are doing it for scheduled jobs count here: https://github.com/scriptnull/waymond/blob/b552d53f85affc019593229b97fa6f92883f4481/internal/trigger/buildkite/buildkite.go#L95-L96

Reply:
BTW, this is how the metrics endpoint output looks like for buildkite: https://buildkite.com/docs/apis/agent-api/metrics (we could probably expose other fields like `waiting` and `total` along with `running` too)